eclass/cargo.eclass

```diff
@@ -76,6 +83,39 @@ ECARGO_VENDOR="${ECARGO_HOME}/gentoo"
 # SRC_URI="${CARGO_CRATE_URIS}"
 # @CODE
 
+# @ECLASS_VARIABLE: CRATE_PATHS_OVERRIDE
+# @DEFAULT_UNSET
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Bash string containing paths to crates that will be used to override
+# dependencies. This is not "smart", _all crates_ which match the
+# `Cargo.toml` in a path will be overridden, ignoring lockfiles,
+# version constraints, etc.
+#
+# This should be used as a last resort where (e.g.) you are
+# bootstrapping Rust and need to override a vendored crate
+# with a newer version, and all versions in use are compatible.
+#
+# Crate names and versions must be separated by a `@`;
+# multiple crates are separated by a space or newline.
+# Crates in CRATE_PATHS_OVERRIDE are implicitly added to CRATES.
```

Comment:
Does this mean that the replaced crates no longer need to be present in `CRATES`?

Reply:
Yes, that is correct. Because we need to state the crates anyway to know which need a config file insertion ( a bit like `GIT_CRATES` without the associative array), it doesn't make sense to duplicate them in `${CRATES}` when we can infer that the package requires them anyway.

I thought about being even more explicit and saying 'Do not include these in `${CRATES}`', however it shouldn't actually hurt anything if they end up in both.

Reply:
I'm just thinking you could leave a note that they can but do not need to be removed.

Reply:
I've updated the comment, but upon reflection that hasn't actually made the situation any better.

The typical use-case for this is ebuilds that don't already have a CRATES variable anyway (e.g. the rust 1.54 ebuild), as it's typically not something packages mix with vendored crates (that I've seen anyway). I'll ensure that the 'How to write a Rust ebuild' docs spell it out though.